src/components/molecules/Monaco/Monaco.tsx

```diff
@@ -223,7 +228,7 @@ const Monaco = (props: {editorHeight: string; diffSelectedResource: () => void;
       <S.HiddenInputContainer>
         <S.HiddenInput ref={hiddenInputRef} type="text" />
       </S.HiddenInputContainer>
-      {firstCodeLoadedOnEditor && (
+      {firstCodeLoadedOnEditor && editingResourceId === selectedResourceId && (
```

Comment:
just trying to understand - how does this fix the problem!?

Reply:
It creates a new editor instance every time`selectedResource` changes. So we create a new undo/redo history for an individual resource. Otherwise, If you select two resource files one after and do undo on the last selected one. The content changes with the previous one.

Reply:
Please try on the main branch and on this branch. You will see the difference. If you can't reproduce it I can show you.